crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -312,162 +475,51 @@ impl BuildSpecCommand {
-					RelayChain::Westend |
-						RelayChain::Paseo | RelayChain::Kusama |
-						RelayChain::Polkadot
-				) {
-					continue;
-				} else {
-					prompt = prompt.item(
-						relay,
-						relay.get_message().unwrap_or(relay.as_ref()),
-						relay.get_detailed_message().unwrap_or_default(),
-					);
-				}
-			},
-		_ =>
-			for relay in RelayChain::VARIANTS {
-				if matches!(
-					relay,
-					RelayChain::Westend |
-						RelayChain::Paseo | RelayChain::Kusama |
-						RelayChain::Polkadot
-				) {
-					continue;
-				} else {
-					prompt = prompt.item(
-						relay,
-						relay.get_message().unwrap_or(relay.as_ref()),
-						relay.get_detailed_message().unwrap_or_default(),
-					);
-				}
-			},
+// Locate binary, if it doesn't exist trigger build.
+fn ensure_binary_exists(
+	cli: &mut impl cli::traits::Cli,
+	mode: &Profile,
+) -> anyhow::Result<PathBuf> {
+	let cwd = current_dir().unwrap_or(PathBuf::from("./"));
+	match binary_path(&mode.target_directory(&cwd), &cwd.join("node")) {
+		Ok(binary_path) => Ok(binary_path),
+		_ => {
+			cli.info("Node was not found. The project will be built locally.".to_string())?;
+			cli.warning("NOTE: this may take some time...")?;
+			build_parachain(&cwd, None, mode, None).map_err(|e| e.into())
+		},
 	}
+}
 
-	let relay_chain = prompt.interact()?.clone();
-
-	// Prompt for default bootnode if chain type is Local or Live.
-	let default_bootnode = match chain_type {
-		ChainType::Development => true,
-		_ => confirm("Would you like to use local host as a bootnode ?".to_string()).interact()?,
-	};
-
-	// Prompt for protocol-id.
-	let default = chain_spec
-		.as_ref()
-		.and_then(|cs| cs.get_protocol_id())
-		.unwrap_or(DEFAULT_PROTOCOL_ID)
-		.to_string();
-	let protocol_id: String = input("Enter the protocol ID that will identify your network:")
-		.placeholder(&default)
-		.default_input(&default)
-		.interact()?;
-
-	// Prompt for genesis state
-	let genesis_state = confirm("Should the genesis state file be generated ?".to_string())
-		.initial_value(true)
-		.interact()?;
-
-	// Prompt for genesis code
-	let genesis_code = confirm("Should the genesis code file be generated ?".to_string())
-		.initial_value(true)
-		.interact()?;
-
-	// Only check user to check their profile selection if a live spec is being built on debug mode.
-	let profile =
-		if !args.release && matches!(chain_type, ChainType::Live) {
-			confirm("Using Debug profile to build a Live specification. Should Release be used instead ?")
-    		.initial_value(true)
-    		.interact()?
-		} else {
-			args.release
-		};
-
-	Ok(BuildSpecCommand {
-		output_file: Some(PathBuf::from(output_file)),
-		release: profile,
-		id: Some(para_id),
-		default_bootnode,
-		chain_type: Some(chain_type),
-		chain: Some(chain),
-		relay: Some(relay_chain),
-		protocol_id: Some(protocol_id),
-		genesis_state,
-		genesis_code,
-	})
+// Prepare the output path provided.
+fn prepare_output_path(mut output_path: PathBuf) -> anyhow::Result<PathBuf> {
```

Comment:
I have noticed that when using `--output ./target/pop/test.json`, like in the integration tests, it will create a directory `test.json` within `target/pop`.

Reply:
Thanks guys! Fixed in [fix: prepare_output_path](https://github.com/r0gue-io/pop-cli/pull/351/commits/efa35baea4308ecae5d84bc199848ff7f0998f75)

Reply:
Nice!